ostd/src/cpu/local/single_instr.rs

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Extensions for CPU-local types that allows single-instruction operations.
+//!
+//! For some per-CPU objects, fetching or modifying the values of them can be
+//! done in a single instruction. Then we would avoid turning off interrupts
+//! when accessing them, which incurs non-trivial overhead.
+//!
+//! This trait is the architecture-specific interface for single-instruction
+//! operations. The architecture-specific module can implement this trait for
+//! common integer types. For architectures that don't support single-
+//! instruction operations, they can provide fallback implementations that
+//! turn off interrupts and preemption.
+//!
+//! Currently we don't plan to implement most of the [`core::ops`] operations
+//! for the sake of simplicity. The status-quos are sufficient enough.
+
+use super::{CpuLocal, __cpu_local_start};
+
+/// An interface for architecture-specific single-instruction add operation.
+pub trait SingleInstructionAddAssign<Rhs = Self> {
+    /// Adds a value to the per-CPU object.
+    ///
+    /// This operation wraps on overflow.
+    ///
+    /// The offset of the object is relative to the base address of the CPU-
+    /// local storage.
+    ///
+    /// # Safety
+    ///
+    /// The caller should ensure that the offset is valid to be written to.
+    /// This operation is not atomic. Accessing the same address from multiple
+    /// threads produces undefined behavior.
+    unsafe fn cpu_local_add_assign(offset: *mut Self, rhs: Rhs);
```

Comment:
By atomic I mean that the implementer doesn't necessarily have to use `lock; mov` to present "sequential consistent" operation. This is the case for x86. For other architectures the acquire-release ordering or even the correctness (for example, sub-word access) is not guaranteed for shared accesses.

As for why it would be shared, I stated it in https://github.com/asterinas/asterinas/pull/1125#issuecomment-2267491546.